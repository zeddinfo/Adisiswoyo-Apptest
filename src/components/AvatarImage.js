import React from 'react'
import { StyleSheet, Text, View } from 'react-native'
import { fonts } from '../styles/fonts'

const AvatarImage = ({ first, second }) => {
    return (
        <View style={styles.container}>
            <Text style={styles.alphabet}>{`${first ?? ''}${second ?? ''}`}</Text>
        </View>
    )
}

export default AvatarImage

const styles = StyleSheet.create({
    container: {
        width: 65,
        height: 65,
        borderRadius: 65 / 2,
        backgroundColor: '#64c5EB',
        alignContent: 'center',
        alignItems: 'center',
        justifyContent: 'center'
    },
    alphabet: {
        fontFamily: fonts.primary[600],
        fontSize: 50,
        color: 'white'
    }
})
