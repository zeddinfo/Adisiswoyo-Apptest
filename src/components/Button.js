import React from 'react'
import { StyleSheet, Text, TouchableOpacity, View } from 'react-native'
import { fonts } from '../styles/fonts'

const Button = ({ title, onPress, }) => {
    return (
        <TouchableOpacity style={styles.container} onPress={onPress} testID='button'>
            <Text style={styles.text}>{title}</Text>
        </TouchableOpacity>
    )
}

export default Button

const styles = StyleSheet.create({
    container: {
        backgroundColor: '#0BCAD4',
        paddingVertical: 10,
        borderRadius: 10,
    },
    text: {
        fontSize: 18,
        fontFamily: fonts.primary[600],
        textAlign: 'center',
        color: 'white'
    }
})
