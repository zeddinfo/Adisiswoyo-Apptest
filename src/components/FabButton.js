import React from 'react'
import { TouchableOpacity } from 'react-native';
import { StyleSheet, Text, View } from 'react-native'
import F5Icon from 'react-native-vector-icons/FontAwesome5';

const FabButton = ({ onPress }) => {
    return (
        <TouchableOpacity style={styles.container} onPress={onPress}>
            <F5Icon name="plus" size={25} color="white" />
        </TouchableOpacity>
    )
}

export default FabButton

const styles = StyleSheet.create({
    container: {
        position: 'absolute',
        bottom: 40,
        right: 50,
        backgroundColor: '#7F58AF',
        width: 60,
        height: 60,
        borderRadius: 60,
        alignItems: 'center',
        alignContent: 'center',
        justifyContent: 'center',
    }
})
