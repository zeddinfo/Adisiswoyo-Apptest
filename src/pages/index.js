import ListContact from "./ListContact";
import CreatePage from './CreatePage';
import EditPage from './EditPage';

export {
    ListContact,
    CreatePage,
    EditPage,
}