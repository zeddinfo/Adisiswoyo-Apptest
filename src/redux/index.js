import store from './store';
import reducer from './reducer';
export * from './actions';

export {
    store,
    reducer,
}