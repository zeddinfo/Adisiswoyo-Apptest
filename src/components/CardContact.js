import React from 'react'
import { Image, StyleSheet, Text, TouchableOpacity, View } from 'react-native'
import { fonts } from '../styles/fonts';
import Gap from './Gap';
import AvatarImage from './AvatarImage';



const CardContact = ({ firstName, lastName, image, age, onPress, id }) => {
    const firstNameAlphabet = firstName.split(' ').pop();
    const lastNameAlphabet = lastName.split(' ').pop();
    // console.log('first', firstNameAlphabet.charAt(0), 'last', lastNameAlphabet.charAt(0));
    // console.log('firstname', firstName);
    return (
        <TouchableOpacity style={styles.container} onPress={onPress} testID='card-test'>
            {(image == '' || image == 'N/A') ? <AvatarImage first={firstNameAlphabet.charAt(0)} /> : <Image source={{ uri: image }} style={styles.image} />}

            <Gap width={20} />
            <View style={styles.profile}>
                <Text style={styles.header} testID='name-title'>{`${firstName} ${lastName}`}</Text>
                <Text style={styles.subHeader} testID='age-title'>{`Age ${age}`}</Text>
            </View>

            {/* <View style={styles.icon}>
                <View style={styles.iconArrow}>
                    <Image source={require('../Assets/icons/ic-arrow-right.png')} style={styles.iconArrow} />
                </View>

            </View> */}
        </TouchableOpacity>
    )
}

export default CardContact

const styles = StyleSheet.create({
    container: {
        backgroundColor: 'white',
        paddingVertical: 20,
        paddingHorizontal: 10,
        marginBottom: 10,
        flexDirection: 'row',
        borderRadius: 4,
        marginHorizontal: 5,
        borderRadius: 20,

    },
    image: {
        width: 65,
        height: 65,
        borderRadius: 65 / 2,
        resizeMode: 'cover'
    },
    header: {
        fontFamily: fonts.primary[600],
        color: 'black'
    },
    subHeader: {
        fontFamily: fonts.primary.normal,
        color: 'black'
    },
    profile: {
        // alignItems: 'center',
        // backgroundColor: 'red',
        justifyContent: 'center',
        alignContent: 'center'
    },

    icon: {
        flex: 1,
        alignItems: 'flex-end',
        justifyContent: 'center',
    },
    iconArrow: {
        width: 15,
        height: 15,

    }
})
