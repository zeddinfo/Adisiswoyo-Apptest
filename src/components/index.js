import HeaderPage from './HeaderPage';
import Gap from './Gap';
import CardContact from './CardContact';
import AvatarImage from './AvatarImage';
import FabButton from './FabButton';
import Search from './Search';
import Header from './Header';
import Input from './Input';
import Button from './Button';
import Profile from './Profile';

export {
    HeaderPage,
    Gap,
    CardContact,
    AvatarImage,
    FabButton,
    Search,
    Header,
    Input,
    Button,
    Profile
}