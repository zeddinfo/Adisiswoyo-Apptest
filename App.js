import React from 'react'
import { StyleSheet, Text, View } from 'react-native'
import { Provider } from 'react-redux'
import AppNavigator from './AppNavigator'
import { store } from './src/redux'

const App = () => {
  return (
    <View style={{ flex: 1, backgroundColor: 'white' }}>
      <Provider store={store}>
        <AppNavigator />
      </Provider>
    </View>
  )
}

export default App

const styles = StyleSheet.create({})
