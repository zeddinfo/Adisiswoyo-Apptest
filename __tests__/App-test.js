
import React from 'react';
import Adapter from 'enzyme-adapter-react-16';
import { shallow, configure } from 'enzyme';
import App from '../App';
import { Button, CardContact, Header, Input } from '../src/components';
import { describe } from '@jest/globals';
import { CreatePage, EditPage, ListContact } from '../src/pages';
import { Provider } from 'react-redux';
import { store } from '../src/redux';
import { Actions } from 'react-native-router-flux';

configure({ adapter: new Adapter(), disableLifecycleMethods: true })

export const findByTestAtrr = (component, attr) => {
  const wrapper = component.find(`[testID='${attr}']`);
  return wrapper;
};

jest.mock('react-native-router-flux', () => ({
  Actions: {
    sceneKey: jest.fn()
  }
}))


describe('App Screen ', () => {
  it('should renders `App Screen` is correctly', () => {
    const wrapper = shallow(<App />);
    expect(wrapper).toMatchSnapshot()
  })
});

describe('List Contact Screen', () => {
  it('should renders `List Contact` is correctly', () => {
    const wrapper = shallow(
      <Provider store={store}>
        <ListContact />
      </Provider>
    );
    expect(wrapper).toMatchSnapshot();
  });

  describe('Render `Card Contacts`', () => {
    let wrapper;
    beforeEach(() => {
      const props = {
        firstName: 'Example',
        lastName: 'Example',
        age: 10,
        photo: 'string'
      }
      wrapper = shallow(<CardContact firstName={props.firstName} lastName={props.lastName} age={props.age} image={props.photo} />)
    });

    it('Should renders without error', () => {
      const component = findByTestAtrr(wrapper, 'card-test');
      expect(component.length).toBe(1);
    });

    it('Should press and navigate to `Detail`', () => {
      const component = findByTestAtrr(wrapper, 'card-test');
      component.simulate('press');
      expect(Actions.sceneKey.mock.calls.legth)
    })
  });

  describe('Detail Screen', () => {
    let wrapper;
    beforeEach(() => {
      wrapper = shallow(<Provider store={store}><EditPage /></Provider>)
    });
    describe('Change Data', () => {
      it('Should change `firstname or last name or age`', () => {
        let input = shallow(<Input />)
        input.find('[testID="input-test"]').simulate('change');
        expect(Actions.sceneKey.mock.calls.legth);
      })
    });
    describe('Update Data', () => {
      it('Should press `Button Update to save data`', () => {
        let button = shallow(<Button />)
        button.find('[testID="button"]').simulate('press');
        expect(Actions.sceneKey.mock.calls.length)
      })
    });
    describe('Delete Data', () => {
      it('Should press `Button Trash to delete data`', () => {
        let button = shallow(<Header />)
        button.find('[testID="delete-button"]').simulate('press');
        expect(Actions.sceneKey.mock.calls.length)
      })
    });

  });

  describe('Created Contact', () => {
    let wrapper;
    beforeEach(() => {
      wrapper = shallow(<Provider store={store}><CreatePage /></Provider>)
    });
    describe('Change Data', () => {
      it('Should change `firstname or last name or age`', () => {
        let input = shallow(<Input />)
        input.find('[testID="input-test"]').simulate('change');
        expect(Actions.sceneKey.mock.calls.legth);
      })
    });
    describe('Save Data', () => {
      it('Should press `Button to save data`', () => {
        let button = shallow(<Button />)
        button.find('[testID="button"]').simulate('press');
        expect(Actions.sceneKey.mock.calls.length)
      })
    });
  })

})



