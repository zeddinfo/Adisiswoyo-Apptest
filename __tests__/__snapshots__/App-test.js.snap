// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Screen  should renders \`App Screen\` is correctly 1`] = `ShallowWrapper {}`;

exports[`App Screen Mount Component should render \`Text Input\` Component correctly 1`] = `ShallowWrapper {}`;

exports[`App Screen Text Input should render \`Text Input\` Component correctly 1`] = `ShallowWrapper {}`;

exports[`App Screen Text Input should render Text Input Component correctly 1`] = `ShallowWrapper {}`;

exports[`App Screen should renders \`App Screen\` is correctly 1`] = `ShallowWrapper {}`;

exports[`App Screen should renders \`App Screen\` module correctly 1`] = `ShallowWrapper {}`;

exports[`List Contact Screen should renders \`List Contact\` is correctly 1`] = `ShallowWrapper {}`;
